### PR TITLE
libClusterFuzz progress.

### PR DESCRIPTION
**Pull request by oliverchang**
- Add test for basic stacktrace parsing.
- Add main module for clusterfuzz.fuzz to allow CLI invocation.
- Add environment.is_lib() to disable certain features that don't make
  sense for the pip package.
- Add config into pip package as a data dependency.

### CODE REVIEW

File: src/python/lib/clusterfuzz/fuzz/__main__.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      type=int,
+      default=3600)
+  args = parser.parse_args()
+
+  # TODO(ochang): Find a cleaner way to propagate this.
+  if args.sanitizer == 'address':
+    os.environ['JOB_NAME'] = 'libfuzzer_asan'
+  elif args.sanitizer == 'memory':
+    os.environ['JOB_NAME'] = 'libfuzzer_msan'
+  elif args.sanitizer == 'memory':
```
Comment (mbarbella-chromium) [inline src/python/lib/clusterfuzz/fuzz/__main__.py:57]: memory -> undefined

**Review by mbarbella-chromium — APPROVED** [review]

File: src/python/lib/clusterfuzz/fuzz/__main__.py (hunk 1 above)
Comment (inferno-chromium) [reply src/python/lib/clusterfuzz/fuzz/__main__.py:57]: +1

File: src/python/lib/clusterfuzz/fuzz/__main__.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    os.environ['JOB_NAME'] = 'libfuzzer_ubsan'
+
+  engine_impl = clusterfuzz.fuzz.get_engine(args.engine)
+  options = engine_impl.prepare(args.corpus, args.target, None)
```
Comment (inferno-chromium) [inline src/python/lib/clusterfuzz/fuzz/__main__.py:61]: use named argument for None

File: src/python/lib/clusterfuzz/fuzz/__main__.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  result = engine_impl.fuzz(args.target, options, args.output,
+                            args.max_duration)
+  print(result.logs)
```
Comment (inferno-chromium) [inline src/python/lib/clusterfuzz/fuzz/__main__.py:64]: anything to print if it succeeded or failed, before we print the logs.

File: src/python/system/environment.py [hunk 4]
```diff
@@ -971,7 +971,7 @@ def set_value(environment_variable, value):
   """Set an environment variable."""
   value_str = str(value)
   environment_variable_str = str(environment_variable)
-  value_str = value_str.replace('%ROOT_DIR%', os.environ['ROOT_DIR'])
+  value_str = value_str.replace('%ROOT_DIR%', os.getenv('ROOT_DIR', ''))
```
Comment (inferno-chromium) [inline src/python/system/environment.py:974]: SHould this be empty string or "."

**Review by inferno-chromium — APPROVED** [review]

File: src/python/lib/clusterfuzz/fuzz/__main__.py (hunk 1 above)
Comment (oliverchang) [reply src/python/lib/clusterfuzz/fuzz/__main__.py:57]: Fixed!

File: src/python/lib/clusterfuzz/fuzz/__main__.py (hunk 2 above)
Comment (oliverchang) [reply src/python/lib/clusterfuzz/fuzz/__main__.py:61]: Actually this shouldn't be none, so fixed that.

File: src/python/lib/clusterfuzz/fuzz/__main__.py (hunk 3 above)
Comment (oliverchang) [reply src/python/lib/clusterfuzz/fuzz/__main__.py:64]: Not sure what the definition of succeeded or failed be should be here. Also think there are other improvements to be made here which will come in future PRs (e.g. handling KeyboardInterrupt, streaming logs).

File: src/python/system/environment.py (hunk 4 above)
Comment (oliverchang) [reply src/python/system/environment.py:974]: For libClusterFuzz, we don't expect ROOT_DIR to be set at all, so have the default of ''.

**Pull request merged**